driver-core/src/main/java/com/datastax/driver/core/HostConnectionPool.java

```diff
@@ -392,7 +407,7 @@ private void maybeSpawnNewConnection() {
     }
 
     void replaceDefunctConnection(final PooledConnection connection) {
-        if (connection.markForTrash.compareAndSet(false, true))
+        if (connection.state.compareAndSet(OPEN, GONE))
             open.decrementAndGet();
         connections.remove(connection);
         connection.closeAsync();
```

Comment:
This currently can't happen because of ConvictionPolicy.addFailure always returning true, but it might be a good idea to only call addConnectionIfUnderMaximum() if the connection.state != TRASH.  It is in theory possible for a trashed connection to be defuncted (channelClosed event for example) and we wouldn't want to add a new connection as result of that.  I might be overthinking that though since this is currently not possible.